lib/absinthe/schema/rule/input_union_input_name_present.ex

```diff
@@ -0,0 +1,45 @@
+defmodule Absinthe.Schema.Rule.InputUnionInputNamePresent do
+  use Absinthe.Schema.Rule
+
+  alias Absinthe.{Schema, Type}
+  require IEx
+
+  @moduledoc false
+
+  def explanation(%{
+        data: %{input_union_type: input_union_type, input_object_type: input_object_type}
+      }) do
+    """
+    The `__inputname` must be defined in a input_object :#{input_object_type} used in input_union :#{
```

Comment:
Rather than this validation, would you want to try and auto-inject the field ?

Reply:
Yah, I'll experiment with that